lib/wpc/wpc.c

```diff
@@ -711,6 +711,89 @@ app_res_e WPC_get_scratchpad_block_max(uint8_t * max_size_p)
     return read_single_byte_msap(MSAP_SCRATCHPAD_BLOCK_MAX, max_size_p);
 }
 
+app_res_e WPC_get_multicast_groups(app_addr_t * addr_list, uint8_t * num_addr_p)
+{
+    app_res_e ret;
+    uint32_t groups[MAXIMUM_NUMBER_OF_MULTICAST_GROUPS];
+
+    // Read multicast groups
+    int res = msap_attribute_read_request(MSAP_MULTICAST_GROUPS,
+                                          4 * MAXIMUM_NUMBER_OF_MULTICAST_GROUPS,
+                                          (uint8_t *) groups);
+
+    ret = convert_error_code(ATT_READ_ERROR_CODE_LUT, res);
+    if (ret != APP_RES_OK)
+    {
+        return ret;
+    }
+
+    for (uint8_t n = 0; n < MAXIMUM_NUMBER_OF_MULTICAST_GROUPS; n++)
+    {
+        if (n < *num_addr_p)
+        {
+            // Fix byte order
+            app_addr_t addr = uint32_decode_le((const uint8_t *) &groups[n]);
+
+            // Convert address to a real multicast address
+            if (addr != 0)
+            {
+                addr |= 0x80000000;
+            }
+
+            // Store address
+            addr_list[n] = addr;
+        }
+    }
+
+    // Set number of addresses found, even if not all could fit in the given buffer
```

Comment:
Shouldn't be an error a better solution?
Or then if you handle it this way, maybe you should skip addr == 0, because if user ask for 5 addresses, it could receive 5 0 addresses while the 5 other ones are set.
Or as it is hard-coded anyway to 10 entries for now, why not asking for a 40 bytes buffers all the time.
Could you be in a system with such constraints that you cannot have a 40 bytes buffer?

Reply:
I wanted the set_*() and get_*() functions to be symmetrical and also very transparent to what actually gets written to the multicast address slots in the node. Therefore I'm not filtering out any zero addresses, but instead leaving them in place. With the API implemented here, it is possible to set or query e.g. just one address, and there is no need to know what the maximum number of addresses is.